### PR TITLE
feat: user profile presigned url

### PR DESCRIPTION
**Pull request by haritianaadriano**

### CODE REVIEW

File: src/main/resources/mock/img.png
Comment (Mahefaa) [inline src/main/resources/mock/img.png:1]: put this in src/test/resources

File: src/main/java/school/hei/haapi/endpoint/rest/controller/TeacherController.java [hunk 1]
```diff
@@ -69,4 +72,11 @@ public List<Teacher> createOrUpdateTeachers(@RequestBody List<CrupdateTeacher> t
         .map(userMapper::toRestTeacher)
         .collect(toUnmodifiableList());
   }
+
+  @PostMapping(value = "/teachers/{id}/picture/raw", consumes = IMAGE_PNG_VALUE)
+  public Teacher uploadTeacherProfilePicture(
+      @RequestBody MultipartFile profilePicture, @PathVariable(name = "id") String teacherId) {
+    userService.uploadUserProfilePicture(profilePicture, teacherId);
+    return userMapper.toRestTeacher(userService.getById(teacherId));
+  }
```
Comment (Mahefaa) [inline src/main/java/school/hei/haapi/endpoint/rest/controller/TeacherController.java]: try it without using multipart

Comment (Mahefaa) [reply src/main/java/school/hei/haapi/endpoint/rest/controller/TeacherController.java]: use byte[] ( technical design not a performance issue)

**Review by Mahefaa — CHANGES_REQUESTED** [review]

File: src/main/resources/mock/img.png
Comment (haritianaadriano) [reply src/main/resources/mock/img.png:1]: and if one day we need other file as mocked file for test ?

**Pull request closed without merging**